src/effects/Phaser.cpp

```diff
@@ -130,6 +130,61 @@ struct EffectPhaser::Validator
 };
 
 
+struct EffectPhaser::Instance
+   : public PerTrackEffect::Instance
+   , public EffectInstanceWithBlockSize
+{
+   explicit Instance(const PerTrackEffect& effect)
+      : PerTrackEffect::Instance{ effect }
+   {}
+
+   bool ProcessInitialize(EffectSettings& settings,
+      double          sampleRate,
+      ChannelNames     chanMap) override;
+
+   size_t ProcessBlock(EffectSettings& settings,
+      const float* const* inBlock, float* const* outBlock, size_t blockLen)  override;
+
+   bool RealtimeInitialize(EffectSettings& settings, double) override;
+
+   bool RealtimeAddProcessor(EffectSettings& settings,
+      unsigned numChannels, float sampleRate) override;
+
+   bool RealtimeFinalize(EffectSettings& settings) noexcept override;
+
+   size_t RealtimeProcess(size_t group, EffectSettings& settings,
+      const float* const* inbuf, float* const* outbuf, size_t numSamples)
+      override;
+
+   unsigned GetAudioInCount() const override;
+   unsigned GetAudioOutCount() const override;
+
+   void InstanceInit(EffectPhaserState& data, float sampleRate);
+
+   size_t InstanceProcess(EffectSettings& settings,
+      EffectPhaserState& data,
+      const float* const* inBlock,
+      float* const* outBlock,
+      size_t                 blockLen);
+
+   void Coefficients(double hz, double slope, double gain, double samplerate, int type,
+      double& a0, double& a1, double& a2, double& b0, double& b1, double& b2);
+
+   float DoFilter(EffectPhaserState& data, float in);
+
+   EffectPhaserState mMaster;
+   std::vector<EffectPhaserState> mSlaves;
```

Comment:
Commenting on the whole PR, it appears to be a correct transformation.  But I want to add a requirement.

Make the objects in this array the same class as the instance -- as is done already for AudioUnits and VST2.

When that works, do the same when you transform other built-in effects to statelessness.  Write PRs to do the same retroactively for the Reverb and Wahwah transformations.

If time allows, make PRs to do the same for Ladspa and LV2.

The reason?

I don't like the duplicated interface for processing EffectInterface.h.  There is RealtimeProcess but also ProcessBlock.  I wabt them unified so the interface is simpler.

That means eliminating the group argument of realtime process.  More about that later:  but making every array of slaves an array of Instances is the beginning.  Then the "array of slaves" is exposed to RealtimeEffectState, not an internal detail of each Instance class.  The RealtimeEffectState will dispatch to the correct Instance object.